internal/cache/theme.go

```diff
@@ -0,0 +1,108 @@
+package cache
+
+import (
+	"strconv"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/internal/svg"
+)
+
+var overrides = make(map[fyne.Widget]*overrideScope)
+
+type overrideScope struct {
+	th      fyne.Theme
+	cacheID string
+}
+
+var nextID = 1
```

Comment:
This seems bound to cause race conditions and potential collisions between ids. Use an atomic counter (use `.Add(1)` and the return value)?

We might need a lock around the map as well maybe?

Reply:
Should all be sorted now, sorry for the re-review